src/analyses/apron/apronAnalysis.apron.ml

```diff
@@ -12,11 +12,12 @@ struct
 
   let name () = "apron"
 
-  module D = ApronComponents (Priv.D)
+  module D = ApronComponents (Priv.AD) (Priv.D)
```

Comment:
Here privatization determines the local domain to be used. This seems backwards. Instead `AD` should be given as argument to `Priv`.

Reply:
At the instantiation site of the Privatization, the `AD` is required as an argument: 
https://github.com/goblint/analyzer/blob/3bd9fa731065542c99ba4302ef6aebab56bfcd82/src/analyses/apron/apronPriv.apron.ml#L682-L698
The passed module is then stored as module `AD` within the privatizations. This module is then retrieved here again. 

If one were to change the quoted location to return a functor from `S2` to `ApronPriv.S`, one would also have to apply the `TracingPriv` functor in the Analysis, because one would not have a `ApronPriv.S` to pass to it in `ApronPriv`. 
Would this be preferable overall?

Reply:
Couldn't one just change `ApronPriv.S` to be `functor (AD: S2) -> sig (* the current signature *) end`? Should still be able to make `TracingPriv` do such lifting.

Reply:
One could alternatively also expose the `ApronComponents` modules of the privatizations, so one would simply use `Priv.ApronComponents` here. But for most of the analysis, one does not really need the whole `ApronComponents` module, but only its type `t`.

Reply:
>Couldn't one just change ApronPriv.S to be functor (AD: S2) -> sig (* the current signature *) end? Should still be able to make TracingPriv do such lifting.

Yes, I'll try it.

Reply:
Done.